### PR TITLE
[docs] Integrate wptserve documentation

### PR DESCRIPTION
**Pull request by jugglinmike**
Relocate instructions that are specifically geared towards test authors
to the "docs" directory. Persist API documentation in its current
location and configure Sphinx to include relevant aspects in the
generated website.

---

The organization of content about "pipes" is a little too focused on implementation details. [Since some of the other documentation also exhibits this problem](https://github.com/bocoup/wpt-docs/issues/3), I suggest we address that issue with more fine-grained restructuring in a follow-up patch.

### CODE REVIEW

File: docs/writing-tests/python-handlers/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Python Handlers
+
+Python file handlers are designed to provide a vaguely PHP-like interface
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]: I strongly dislike calling this "vaguely PHP-like".

File: docs/writing-tests/python-handlers/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+where each resource corresponds to a particular python file on the
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]:
```suggestion
where each resource corresponds to a particular Python file on the
```

File: docs/writing-tests/python-handlers/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+filesystem. Typically this is hooked up to a route like ``("*",
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]: Let's not say that's "typically"; for WPT it *is*.

File: docs/writing-tests/python-handlers/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+"*.py", python_file_handler)``, meaning that any .py file will be
+treated as a handler file (note that this makes python files unsafe in
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]:
```suggestion
treated as a handler file (note that this makes Python files unsafe in
```

File: docs/writing-tests/python-handlers/index.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+much the same way that .php files are when using PHP).
+
+Unlike PHP, the python files don't work by outputting text to stdout
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]: Let's not quite so directly reference PHP (which isn't so useful if you don't know what PHP does!), but instead say something more like, "These Python files are assumed to define a single function `main(request, response)`. Note therefore anything written to stdout is not returned as part of the response, as is the case in PHP.

**Review by gsnedders — CHANGES_REQUESTED** [review]
This seems wholly like a good move to me.

**Comment by jugglinmike** [comment]
> This seems wholly like a good move to me.

:)

The content of `docs/writing-tests/python-handlers/index.md` was all cribbed from `tools/wptserve/docs/handlers.rst`. That doesn't make your feedback any less beneficial, but it does inform how I'd like to format the patch.

I started fresh with a commit to improve the existing documentation as per your recommendation. Then, I rebased my original patch over the new commit. This separates the content change from the refactoring, limiting the subjective aspects of the latter and improving the discoverability of the former.

[The previous version of this branch is available here.](https://github.com/web-platform-tests/wpt/compare/master...bocoup:docs-sphinx-3-wptserve-orig)

File: docs/writing-tests/python-handlers/index.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Python Handlers
+
+Python file handlers are Python files which the server executes in response to
+requests made to the corresponding URL. This is hooked up to a route like
+`("*", "*.py", python_file_handler)`, meaning that any .py file will be
+treated as a handler file (note that this makes it easy to write unsafe
+handlers, particularly when running the server in a web-exposed setting).
+
+The Pyton filess must define a single function `main` with the signature::
```
Comment (gsnedders) [inline docs/writing-tests/python-handlers/index.md]:
```suggestion
The Python files must define a single function `main` with the signature::
```

**Review by gsnedders — APPROVED** [review]
Apart from this one typo, LGTM; sorry for making you fix something I didn't realise was existing content!

**Comment by jugglinmike** [comment]
No worries. They were good suggestions, and this was a good time to implement them.

Happy to squash when you think this is ready.

**Comment by gsnedders** [comment]
@jugglinmike 
> Happy to squash when you think this is ready.

now

**Comment by jugglinmike** [comment]
Good to go, @gsnedders

**Pull request merged**